### PR TITLE
feat(gateway): google-vertex embeddings support

### PR DESCRIPTION
**Pull request by RATCHAW**
## Summary

Adds `google-vertex` as a provider for `gemini-embedding-001` and `gemini-embedding-2` on `/v1/embeddings`, mirroring the AI Studio support from #2305: `gemini-embedding-001` routes via Vertex `:predict`, `gemini-embedding-2` via v1beta1 `:embedContent`, batched inputs fan out into N parallel calls combined into one OpenAI-shaped response. Auth uses an OAuth2 access token minted from a service-account JSON (Vertex's PredictionService rejects Express-mode API keys), reusing the existing `getVertexOpenAIAccessToken` helper. `findEmbeddingMapping` now accepts a `provider/model` prefix so callers can disambiguate when a model id is offered by multiple providers, the `LLM_<PROVIDER>_BASE_URL` env override is honored for both `google-*` providers, and OAuth-exchange failures return a normalized `502 vertex_oauth_failed` with key-health tracking instead of an opaque 500.

## Test plan

- [x] `pnpm format` + `pnpm build` clean
- [x] New e2e tests in `api.spec.ts` cover single/batched inputs, `dimensions`, base64 encoding, token-ID rejection, missing project id, missing/invalid service-account, OAuth-exchange failure, and the v2 `:embedContent` shape

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added Google Vertex AI as an embeddings provider with provider-prefixed model routing and optional base64-encoded embeddings; supports both service-account and short-lived OAuth flows.

* **Bug Fixes / Validation**
  * Enforced single-input for Gemini embeddings and improved errors for missing project ID, invalid credentials, unsupported input types, token-exchange failures, and routing edge cases.

* **Tests**
  * Expanded end-to-end coverage and strengthened Vertex mocking behavior.

* **Documentation**
  * Updated embeddings docs with Vertex model IDs, batching notes, and usage guidance.

[Review Change Stack](https://app.coderabbit.ai/change-stack/theopenco/llmgateway/pull/2332?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: apps/gateway/src/embeddings/embeddings.ts [hunk 1]
```diff
@@ -602,18 +622,32 @@ embeddings.openapi(createEmbeddings, async (c): Promise<any> => {
...
 	};
+	// Env baseUrl override: LLM_<PROVIDER>_BASE_URL can redirect upstream
+	// traffic to proxies, regional endpoints, or test mocks. Applies even
+	// with a provider key set — providerKey.baseUrl still wins via the ??
+	// chain below, so BYOK callers can opt out by setting their own
+	// baseUrl. Only google-* providers expose a baseUrl env in
+	// packages/models/src/providers.ts.
+	const envBaseUrl =
+		providerId === "google-vertex" || providerId === "google-ai-studio"
+			? getProviderEnvValue(providerId, "baseUrl", configIndex)
```
Comment (steebchen) [inline apps/gateway/src/embeddings/embeddings.ts]: this should be applicable to all providers including openai @RATCHAW

File: apps/docs/content/features/embeddings.mdx [hunk 2]
```diff
@@ -16,8 +16,13 @@ Browse available embedding models on the [models page](https://llmgateway.io/mod
 
 - **OpenAI** — `text-embedding-3-small`, `text-embedding-3-large`, `text-embedding-ada-002`
 - **Google AI Studio** — `gemini-embedding-2` (recommended), `gemini-embedding-001` (legacy)
+- **Google Vertex AI** — `gemini-embedding-2`, `gemini-embedding-001`. Requires:
+  - A **service-account JSON** for `LLM_GOOGLE_VERTEX_API_KEY` (or as the provider-key token). Vertex's `:predict` and `:embedContent` endpoints reject Express-mode API keys (`API_KEY_SERVICE_BLOCKED` / `CREDENTIALS_MISSING`), so the gateway mints a short-lived OAuth2 access token from the SA JSON for embedding requests.
```
Comment (steebchen) [inline apps/docs/content/features/embeddings.mdx]: @RATCHAW let's revert this documentation diff if not applicable

File: apps/docs/content/features/embeddings.mdx [hunk 3]
```diff
@@ -16,6 +16,7 @@ Browse available embedding models on the [models page](https://llmgateway.io/mod
 
 - **OpenAI** — `text-embedding-3-small`, `text-embedding-3-large`, `text-embedding-ada-002`
 - **Google AI Studio** — `gemini-embedding-2` (recommended), `gemini-embedding-001` (legacy)
+- **Google Vertex AI** — `gemini-embedding-001`, `text-embedding-005`, `text-embedding-004`, `text-multilingual-embedding-002`. Prefix the model id with `google-vertex/` (e.g. `google-vertex/gemini-embedding-001`) when the same id also exists on `google-ai-studio`. `gemini-embedding-001` accepts only one input per request on Vertex; the `text-embedding-*` family supports native batching.
```
Comment (steebchen) [inline apps/docs/content/features/embeddings.mdx]: let's not mention this, it's just confusing. we should just have 1-2 models as an example, and then link to the /models with the embedding filter toggled so users have correct information and the docs don't get outdated too quickly.
